contracts/vaults/Claim.sol

```diff
@@ -33,43 +33,58 @@ contract Claim is Base {
         string calldata _descriptionHash)
     external
     onlyCommittee(_pid)
-    noSubmittedClaims(_pid) {
+    noSubmittedClaims(_pid)
+    {
         require(_beneficiary != address(0), "HVE04");
         // require we are in safetyPeriod
         // solhint-disable-next-line not-rely-on-time
         require(block.timestamp % (generalParameters.withdrawPeriod + generalParameters.safetyPeriod) >=
         generalParameters.withdrawPeriod, "HVE05");
         require(_bountyPercentage <= bountyInfos[_pid].maxBounty, "HVE06");
-
-        submittedClaims[_pid] = SubmittedClaim({
+        uint256 claimId;
+        uint256 nonce;
+        do
```

Comment:
I see what you are doing but I think it is too complex.
We will only have a collision with ids if **within a single block** the following happens:

- a claim is submitted and gets id keccak(poolId, block.number)
- that claim is resolved
- a new claim is submitted <- this would get the same id as the first claim

IT is almost impossible to have this situation, and if it occurs probably something went wrong. So I do not think we need to write code especially to resolve this situation.

Instead, I'd suggest:
```
claimId = keccack256(_pid, block.number)
require(submittedClaims[claimId].beneficiary != 0, "a claim was already submitted in thsi block")
```

Reply:
that will actually be totally impossible with the challenge period.. i was thinking more about hash collisions, almost impossible as they may be. that's why i was checking for 0 too

Reply:
ah no we do not worry about hash collisions
in theory the challenge period can be set to 0, and so, again theoretcially, I think my scenario could happen, right?

Reply:
hmm even when it's set to 0 the submitting of a claim and approving/dismissing can't happen in the same block because they have to see the claim before approving/dismissing and it can't happen in the same time

Reply:
but hash collisions could theoretically happen :)